lib/screens/login/login.dart

```diff
@@ -39,40 +46,182 @@ class LoginScreen extends StatelessWidget {
   }
 
   Widget _buildInitial(BuildContext context) {
-    return Form(
-      key: _formKey,
-      child: Padding(
-        padding: EdgeInsets.symmetric(horizontal: 20.0),
-        child: Column(
-          mainAxisAlignment: MainAxisAlignment.center,
-          crossAxisAlignment: CrossAxisAlignment.center,
-          children: <Widget>[
-            Padding(
-              padding: EdgeInsets.only(bottom: 20.0),
-              child: EmailField(emailController),
-            ),
-            Padding(
-              padding: EdgeInsets.only(bottom: 20.0),
-              child: PasswordField(passwordController),
-            ),
-            Padding(
-              padding: EdgeInsets.only(bottom: 20.0),
-              child: RaisedButton(
-                onPressed: () => _login(context),
-                child: Text("Login"),
-              ),
-            ),
-            Padding(
-              padding: EdgeInsets.all(20.0),
-              child: RaisedButton(
-                onPressed: () {
-                  print("BUTTON PRESS");
-                },
-                child: Text("SignUp"),
+    double width = MediaQuery.of(context).size.width;
+    double height = MediaQuery.of(context).size.height;
+    double bottom = MediaQuery.of(context).viewInsets.bottom;
+    double heightFactor = height / 1000;
+    if (_scrollController.hasClients) {
+      if (bottom > height * 0.25) {
+        _scrollController.animateTo(
+          bottom - height * 0.25,
+          duration: Duration(milliseconds: 300),
+          curve: Curves.ease,
+        );
+      } else {
+        _scrollController.animateTo(0, duration: Duration(milliseconds: 300), curve: Curves.ease);
+      }
+    }
+    return DefaultTextStyle(
+      style: GoogleFonts.roboto().copyWith(color: C.primaryUnHighlightedColor),
+      child: Stack(
+        children: [
+          //Handles background elements
+          ScreenBackground(
+            elementId: 1,
+          ),
+          SingleChildScrollView(
+            physics: NeverScrollableScrollPhysics(),
+            controller: _scrollController,
+            child: Container(
+              height: height,
+              child: Column(
+                mainAxisSize: MainAxisSize.min,
+                children: [
+                  //Space for clouds
+                  Expanded(
+                    child: Container(),
+                    flex: 2,
+                  ),
```

Comment:
Attempt to reduce number of lines, wherever possible.
Also, everywhere place flex as the first parameter, instead of second.
```suggestion
                  Expanded(flex: 2, child: Container()),
```